### ISSUE TITLE
`parallels-iso` builder in Packer 0.9.0+ could not build Mac OS X boxes

### ISSUE THREAD

**Issue by legal90** [mitchellh/packer#4248]
TL;DR: Starting from Packer 0.9.0 `parallels-iso` builder could not build Mac OS X boxes if `iso_url` points to a remote source.

## Details:
Unlikely other guest operating systems, Mac OS X should be installed from the DMG bootable disk image, not an ISO. But it doesn't make sense for Packer, so we can freely use `*[virtualbox|vmware|parallels]-iso` builders to build such boxes.

However, in Parallels Desktop there is an important thing:
If we want to boot the Mac OS X virtual machine from DMG image, then this image file should be named with `.dmg` extension. Otherwise, the emulated EFI fails to parse the partition table and the VM won't boot from this image. 

So, there is no any problem if we set `iso_url` to a local file:
```
packer build -only parallels-iso \
  -var 'iso_url=/path/to/local/Install_macOS.dmg' \
  macos-10.12.json
```
This DMG will be attached to the virtual DVD-ROM as is. 

But if we set the http url, like this:
```
packer build -only parallels-iso \
  -var 'iso_url=http://example.com/download/Install_macOS.dmg' \
  macos-10.12.json
```
then Packer will download this file and saves it to cache with the name ending with `.iso`, so the image became non-bootable for Parallels Desktop. It is hard-coded in all `*-iso` builders: [link](https://github.com/mitchellh/packer/blob/74840e29c30482dd440da99a45f4f1692e4d9f31/builder/parallels/iso/builder.go#L142)

## Suggestion:
That behavior with forced `.iso` was introduced in Packer 0.9.0  by https://github.com/mitchellh/packer/pull/2849 and it looks reasonable for cases like https://github.com/mitchellh/packer/issues/1839.

So, I think that it make sense to manage this `Extention` value by additional configuration option. I'm not sure - should it be implemented in `parallels-iso` only, or in other builders as well. Any suggestions?

cc: @rickard-von-essen

**Comment by rickard-von-essen** [comment]
IMHO it's best to add it in `common.ISOConfig` and make it available to all local builders.

**Comment by legal90** [comment]
Thanks, I'll manage to submit a PR soon.
P.s. Spelling fixed ;)
